common/src/main/java/org/jboss/hal/testsuite/util/AvailablePortFinder.java

```diff
@@ -0,0 +1,173 @@
+package org.jboss.hal.testsuite.util;
+
+import java.io.IOException;
+import java.net.DatagramSocket;
+import java.net.ServerSocket;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Finds currently available server ports.
+ *
+ * @author <a href="http://mina.apache.org">Apache MINA Project</a>
+ * @see <a href="http://www.iana.org/assignments/port-numbers">IANA.org</a>
+ */
+public class AvailablePortFinder {
+    /**
+     * The minimum number of user port number.
+     */
+    public static final int MIN_PORT_NUMBER = 1024;
+
+    /**
+     * The maximum number of user port number.
+     */
+    public static final int MAX_PORT_NUMBER = 49151;
+
+    /**
+     * Creates a new instance.
+     */
+    private AvailablePortFinder() {
+        // Do nothing
+    }
+
+    /**
+     * Returns the {@link Set} of currently available port numbers
+     * ({@link Integer}).  This method is identical to
+     * <code>getAvailablePorts(MIN_PORT_NUMBER, MAX_PORT_NUMBER)</code>.
+     *
+     * WARNING: this can take a very long time.
+     */
+    private static Set<Integer> getAvailablePorts() {
+        return getAvailablePorts(MIN_PORT_NUMBER, MAX_PORT_NUMBER);
+    }
+
+    /**
+     * Gets an available port, selected by the system.
+     *
+     * @throws NoSuchElementException if there are no ports available
+     */
+    private static int getNextAvailable() {
+        ServerSocket serverSocket = null;
+
+        try {
+            // Here, we simply return an available port found by the system
+            serverSocket = new ServerSocket(0);
+            int port = serverSocket.getLocalPort();
+
+            // Don't forget to close the socket...
+            serverSocket.close();
+
+            return port;
+        } catch (IOException ioe) {
+            throw new NoSuchElementException(ioe.getMessage());
+        }
+    }
+
+    /**
+     * Gets the next available port starting at a port.
+     *
+     * @param fromPort the port to scan for availability
+     * @throws NoSuchElementException if there are no ports available
+     */
+    private static int getNextAvailable(int fromPort) {
+        if (fromPort < MIN_PORT_NUMBER || fromPort > MAX_PORT_NUMBER) {
+            throw new IllegalArgumentException("Invalid start port: " + fromPort);
+        }
+
+        for (int i = fromPort; i <= MAX_PORT_NUMBER; i++) {
+            if (available(i)) {
+                return i;
+            }
+        }
+
+        throw new NoSuchElementException("Could not find an available port " + "above " + fromPort);
+    }
+
+    /**
+     * Checks to see if a specific port is available.
+     *
+     * @param port the port to check for availability
+     */
+    private static boolean available(int port) {
```

Comment:
availability is not very good name. You can be looking for it from different point of views and from one client PoV available can be understood as port is in use and is available to connect to it remotely. I suggest using more descriptive name to prevent confusion.